statline_bq/utils.py

```diff
@@ -687,21 +607,19 @@ def convert_table_to_parquet(
     pq_path = Path(f"{out_dir}/{file_name}.parquet")
 
     # Dump each bag partition to json file
-    bag.map(json.dumps).to_textfiles(temp_json_dir / "*.json")
+    print(f"Dumping bag to textfiles for {file_name}")
+    bag.map(ndjson.dumps).to_textfiles(temp_json_dir / "*.json")
```

Comment:
where is `temp_json_dir` defined?

Reply:
It is defined a couple of lines above, as `temp_json_dir = out_dir.parent / Path(f"json/{file_name}")`, where `out_dir` and `file_name` are parameters.  I'll make the handover more explicit, when calling the function: `convert_table_to_parquet(bag=table, file_name=table_name, out_dir=pq_dir, odata_version=odata_version)`

`pq_dir` is created earlier in the flow, through `pq_dir = create_named_dir(id=id, odata_version=odata_version, source=source, config=config)`, which uses `config.paths.temp` as root.

So, the actual file structure (if not changing the config file) is:
```
.
|── AUTHORS.md
|── CONTRIBUTING.md
|── LICENSE
|── Makefile
|── README.md
|── docs
|── poetry.lock
|── pyproject.toml
|── statline_bq
│   |── cli.py
│   |── config.py
│   |── config.toml
│   |── datasets.toml
│   |── temp
│   │   └── cbs
│   │       |── v3
│   │       │   └── 83583NED
│   │       │          └── [UPLOAD-DATE]
│   │       │               └── json
│   │       │                   └── cbs.v3.83583NED_[table_1] <- folders for this table's jsons from dask.bag
│   │       │                   └── cbs.v3.83583NED_[table_2]
│   │       │                   └── etc...
│   │       │               └── parquet <- This is the folder that is eventually uploaded to GCS
│   │       │                   └── cbs.v3.83583NED_[table_1].parquet
│   │       │                   └── cbs.v3.83583NED_[table_2].parqeut
│   │       │                   └── etc...
│   │       └── v4
│   │           └── 83765NED
│   │       │          └── [UPLOAD-DATE]
│   │       │               └── json
│   │       │               └── parquet
│   └── utils.py
└── tests
    |── __init__.py
   └── test_statline_bq.py
```

Reply:
`temp_json_dir` is defined a couple of lines above, through `temp_json_dir = out_dir.parent / Path(f"json/{file_name}")`. `out_dir` and `file_name` are passed as parameters.

`out_dir` is `pq_dir`, and `file_name` is `table_name`, I've made that more explicit when calling the function:
```python
pq_path = convert_table_to_parquet(
                bag=table,
                file_name=table_name,
                out_dir=pq_dir,
                odata_version=odata_version,
            )
```

`pq_dir` is determined eventually in `config.toml`, using `config.paths.temp` and `config.paths.cbs` (generally  `config.paths.[source]`), to create the following structure:
```bash
.
├── AUTHORS.md
├── CONTRIBUTING.md
├── LICENSE
├── Makefile
├── README.md
├── poetry.lock
├── pyproject.toml
├── statline_bq
│   ├── cli.py
│   ├── config.py
│   ├── config.toml
│   ├── datasets.toml
│   ├── temp
│   │   ├── cbs
│   │   │   ├── v3
│   │   │   │   └── 83583NED
│   │   │   │       └── 20210226
│   │   │   │           ├── json # NOTE: Each folder nested here is used by bag.to_textfiles()
│   │   │   │           │   ├── cbs.v3.83583NED_Bedrijfsgrootte
│   │   │   │           │   ├── cbs.v3.83583NED_BedrijfstakkenBranchesSBI2008
│   │   │   │           │   ├── cbs.v3.83583NED_CategoryGroups
│   │   │   │           │   ├── cbs.v3.83583NED_DataProperties
│   │   │   │           │   ├── cbs.v3.83583NED_Perioden
│   │   │   │           │   └── cbs.v3.83583NED_TypedDataSet
│   │   │   │           └── parquet # NOTE: This is the folder that is uploaded to GCS
│   │   │   │               ├── cbs.v3.83583NED_Bedrijfsgrootte.parquet
│   │   │   │               ├── cbs.v3.83583NED_BedrijfstakkenBranchesSBI2008.parquet
│   │   │   │               ├── cbs.v3.83583NED_CategoryGroups.parquet
│   │   │   │               ├── cbs.v3.83583NED_ColDescriptions.json
│   │   │   │               ├── cbs.v3.83583NED_DataProperties.parquet
│   │   │   │               ├── cbs.v3.83583NED_Metadata.json
│   │   │   │               ├── cbs.v3.83583NED_Perioden.parquet
│   │   │   │               └── cbs.v3.83583NED_TypedDataSet.parquet
│   │   │   └── v4
│   │   │       └── 83765NED
│   │   │           └── 20210226
│   │   │               ├── json # NOTE: Each folder nested here is used by bag.to_textfiles()
│   │   │               │   ├── cbs.v4.83765NED_Dimensions
│   │   │               │   ├── cbs.v4.83765NED_MeasureCodes
│   │   │               │   ├── cbs.v4.83765NED_MeasureGroups
│   │   │               │   ├── cbs.v4.83765NED_Observations
│   │   │               │   ├── cbs.v4.83765NED_WijkenEnBuurtenCodes
│   │   │               │   └── cbs.v4.83765NED_WijkenEnBuurtenGroups
│   │   │               └── parquet # NOTE: This is the folder that is uploaded to GCS
│   │   │                   ├── cbs.v4.83765NED_ColDescriptions.json
│   │   │                   ├── cbs.v4.83765NED_Dimensions.parquet
│   │   │                   ├── cbs.v4.83765NED_MeasureCodes.parquet
│   │   │                   ├── cbs.v4.83765NED_MeasureGroups.parquet
│   │   │                   ├── cbs.v4.83765NED_Metadata.json
│   │   │                   ├── cbs.v4.83765NED_Observations.parquet
│   │   │                   ├── cbs.v4.83765NED_WijkenEnBuurtenCodes.parquet
│   │   │                   └── cbs.v4.83765NED_WijkenEnBuurtenGroups.parquet
│   │   └── mlz
│   │       └── v3
│   │           └── 40060NED
│   └── utils.py
└── tests
    ├── __init__.py
    └── test_statline_bq.py

```

I think the general setup and nesting is decent (comments are welcome), but would question:
1. Should we place this entire file structure else where? Currently it is nested within the library itself (meaning within `statline_bq/statline-bq/`) - not sure what is customary or if it matters.
2. As we've previously discussed when creating the `config` item, adding a new source requires manually adding an entry in `config.toml` under `[paths]`. If we wanted to, we could probably automate that. I don't think I would bother with it at this point, though.

Reply:
temp directory should never go in the source code directory. I propose to stick to the Linux standard:

> In Unix and Linux, the global temporary directories are /tmp and /var/tmp. Web browsers periodically write data to the tmp directory during page views and downloads. Typically, /var/tmp is for persistent files (as it may be preserved over reboots), and /tmp is for more temporary files.

Reply:
Check. Python provides `tempfile.gettempdir()` - which is cross platform. Handy for development, especially since my local machine is a mac.